hyperspy/misc/utils.py

```diff
@@ -944,9 +944,9 @@ def create_map_objects(function, nav_size, iterating_kwargs, **kwargs):
     Moved to a separate method to reduce code duplication.
     """
     from hyperspy.signal import BaseSignal
-    from itertools import repeat
+    from itertools import repeat, cycle
 
-    iterators = tuple(signal[1]._iterate_signal()
+    iterators = tuple(cycle(signal[1]._iterate_signal())
```

Comment:
This indeed meets the requirements, but also might require significant additional memory (since the in-built `cycle` just stores the yielded results).

Have you considered adding a new method, for example `_cycle_signal` to do exactly this, but without the drawback? It could be very similar to the `_iterate_signal`, but with a different end condition - for one would stop, and the other would start at the beginning. In fact, I would hope the two methods to be almost identical, and thus it might be possible to refactor them in a very pleasant manner :)